modules/parrableIdSystem.js

```diff
@@ -23,11 +23,36 @@ function isValidConfig(configParams) {
   return true;
 }
 
+function serializeParrableId(idObj) {
+  if (!idObj.eid) return '';
+
+  let str = 'eid:' + idObj.eid;
+  if (idObj.ibaOptout) {
+    str += ',ibaOptout:1';
+  }
+  if (idObj.ccpaOptout) {
+    str += ',ccpaOptout:1';
+  }
+  return encodeURIComponent(str);
+}
+
+function deserializeParrableId(value) {
+  const idObj = {};
+  const values = decodeURIComponent(value).split(',');
+
+  values.forEach(function(value) {
+    const obj = value.split(':');
+    idObj[obj[0]] = +obj[1] === 1 ? true : obj[1];
+  });
+
+  return idObj;
+}
+
 function fetchId(configParams, consentData, currentStoredId) {
   if (!isValidConfig(configParams)) return;
 
   const data = {
-    eid: currentStoredId || null,
+    eid: currentStoredId ? deserializeParrableId(currentStoredId).eid : null,
```

Comment:
In thinking this through more, accessing `.eid` is still unsafe if we have a currentStoredId that's just `ccpaOptout:1`. We would not get an eid property.